Jenkinsfile

```diff
@@ -0,0 +1,82 @@
+pipeline {
+    agent none
+    environment {
+        projectName = 'DDF'
+    }
+    stages {
+        stage('Setup') {
+            steps{
+                notifySlack('good', "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}")
+            }
+        }
+        stage('Parallel Build') {
+            // TODO refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
+            steps{
+                parallel(
+                    linux: {
+                        node('linux-small') {
+                            timeout(time: 3, unit: 'HOURS') {
+                                checkout scm
+                                withMaven(maven: 'M3', mavenSettingsConfig: 'codice-maven-settings') {
+                                    sh 'mvn clean install'
+                                }
+                            }
+                        }
+                    }, windows: {
+                        node('proxmox-windows'){
+                            timeout(time: 3, unit: 'HOURS') {
+                                checkout scm
+                                withMaven(maven: 'M3', mavenSettingsConfig: 'codice-maven-settings') {
+                                    bat 'mvn clean install'
+                                }
+                            }
+                        }
+                    }
+                )
+            }
+        }
+        stage('Static Analysis') {
+            agent { label 'linux-small' }
+            steps{
+                timeout(time: 2, unit: 'MINUTES') {
+                    sh """echo Static Analysis run of "$projectName" """
+                    //TODO DDF-2946 Add SA tools
+                }
+            }
+            post {
+                failure {
+                    sh """Static Analysis failure"""
+                }
+            }
+        }
+        stage('Deploy') {
+            agent { label 'linux-small' }
```

Comment:
I realize there is nothing really in this method but right now since this is done with a separate agent, a deploy would have to rebuild everything that was previously built in the "Parallel Build" section